boards/arm/adafruit_feather_m0_basic_proto/CMakeLists.txt

```diff
@@ -1,6 +1,6 @@
 # SPDX-License-Identifier: Apache-2.0
 
 if(CONFIG_PINMUX_SAM0)
-  zephyr_library()
+  zephyr_library_named(pinmux)
```

Comment:
just one comment: why do we need to _name_ the library in this case? Is name relevant? The board `pinmux.c` files are not part of the pinmux driver/library, they are board related files that setup pinmux at system init.

Reply:
that depends on how we want to solve this issue.

If creating a common library for pinmux functionality, then the best is to have a known name so that we can refer to the same library when appending files to it from different folders.